localstack-core/localstack/services/stepfunctions/asl/utils/boto_client.py

```diff
@@ -1,19 +1,42 @@
+from typing import Optional
+
 from botocore.client import BaseClient
 from botocore.config import Config
 
 from localstack.aws.connect import connect_to
 from localstack.services.stepfunctions.asl.component.common.timeouts.timeout import TimeoutSeconds
+from localstack.services.stepfunctions.asl.component.state.state_execution.state_task.credentials import (
+    ComputedCredentials,
+    Credentials,
+)
+from localstack.utils.aws.client_types import ServicePrincipal
+
+_BOTO_CLIENT_CONFIG = config = Config(
+    parameter_validation=False,
+    retries={"max_attempts": 0, "total_max_attempts": 1},
```

Comment:
nit: Setting both is `max_attempts` and `total_max_attempts` is redundant. Consider rather just `"total_max_attempts": 1`